### PR TITLE
Allow setting collections using a function

### PR DESCRIPTION
**Pull request by leopoletto**
This PR addresses the issue https://github.com/tighten/jigsaw/issues/572

It allows us to set the collection using a function.

It means that now it is possible to get the collections and entries from an external API dynamically without having to expressly set the collections.

### CODE REVIEW

File: tests/TestCase.php [hunk 1]
```diff
@@ -108,6 +108,13 @@ public function buildSite($vfs, $config = [], $pretty = false, $viewPath = '/sou
     {
         $this->app->consoleOutput->setup($verbosity = -1);
         $this->app->config = collect($this->app->config)->merge($config);
+
+        if ($collections = value($this->app->config->get('collections'))) {
+            $this->app->config->put('collections', collect($collections)->flatMap(function ($value, $key) {
+                return is_array($value) ? [$key => $value] : [$value => []];
+            }));
+        }
```
Comment (bakerkretzmar) [inline tests/TestCase.php:116]: Why is this necessary?

**Review by leopoletto — COMMENTED** [review]
Revert all these whitespace/formatting changes please.

**Comment by leopoletto** [comment]
@bakerkretzmar Sorry for not answering and fixing it. Thanks for merging. 💯
